strawberryfields/tdm/tdmprogram.py

```diff
@@ -204,6 +210,7 @@ def reshape_samples(all_samples, modes, N, timebins):
             # create an entry for the new mode with a nested list for each timebin
             new_samples[mode_idx] = [[] for _ in range(timebins)]
 
+        # pops samples, thus emptying all_samples for memory conservation
         sample = all_samples[mode].pop(0)[0]
```

Comment:
another trivial way to solve this without having to copy `all_samples` would be to do
```suggestion
        sample = all_samples[mode][index][0]
        index += 1
```
with index being initialized to zero somewhere at the beginning. Was the only reason not to do that the hassle of managing the index?

Reply:
Oh, true. That's actually quite a bit nicer. Not sure why I didn't think of this before. 😆 Thanks @lneuhaus. I might just have a go at fixing it like this instead.